packages/zk-eddsa-event-ticket-pcd/src/ZKEdDSAEventTicketPCD.ts

```diff
@@ -0,0 +1,538 @@
+import { groth16 } from "snarkjs";
+import { v4 as uuid } from "uuid";
+import vkey from "../artifacts-unsafe/verification_key.json";
+
+import { ZKEdDSAEventTicketCardBody } from "./CardBody";
+
+export const STATIC_TICKET_PCD_NULLIFIER = generateSnarkMessageHash(
+  "dummy-nullifier-for-eddsa-event-ticket-pcds"
+);
+
+export const ZKEdDSAEventTicketPCDTypeName = "zk-eddsa-event-ticket-pcd";
+
+let initializedPromise: Promise<void> | undefined;
+let babyJub: BabyJub;
+let eddsa: Eddsa;
+let initArgs: ZKEdDSAEventTicketPCDInitArgs | undefined = undefined;
+
+/**
+ * Specifies which fields of an EdDSATicket should be revealed in a proof.
+ */
+export interface EdDSATicketFieldsToReveal {
+  revealTicketId?: boolean;
+  revealEventId?: boolean;
+  revealProductId?: boolean;
+  revealTimestampConsumed?: boolean;
+  revealTimestampSigned?: boolean;
+  revealAttendeeSemaphoreId?: boolean;
+  revealIsConsumed?: boolean;
+  revealIsRevoked?: boolean;
+}
+
+/**
+ * Info required to initialize this PCD package.  These are the artifacts
+ * associated with the circom circuit.
+ */
+export interface ZKEdDSAEventTicketPCDInitArgs {
+  zkeyFilePath: string;
+  wasmFilePath: string;
+}
+
+/**
+ * Max supported size of validEventIds field in ZKEdDSAEventTicketPCDArgs.
+ */
+export const VALID_EVENT_IDS_MAX_LEN = 100;
+
+/**
+ * Arguments to request a new proof.
+ */
+export interface ZKEdDSAEventTicketPCDArgs {
+  // generally, `ticket` and `identity` are user-provided
+  ticket: PCDArgument<EdDSATicketPCD>;
+  identity: PCDArgument<SemaphoreIdentityPCD>;
+
+  // `validEventIds` is usually app-specified.  It is optional, and if included
+  // the PCD proves that the ticket's event ID is in this list.  This is a list of
+  // UUIDs with max length VALID_EVENT_IDS_MAX_LEN (100).
+  validEventIds?: StringArrayArgument;
+
+  // `fieldsToReveal`, `externalNullifier`, `watermark` are usually app-specified
+  fieldsToReveal: ObjectArgument<EdDSATicketFieldsToReveal>;
+  watermark: BigIntArgument;
+
+  // provide externalNullifier field to request a nullifierHash
+  // if you don't provide this field, no nullifierHash will be outputted
+  externalNullifier?: BigIntArgument;
+}
+
+/**
+ * Claim part of a ZKEdDSAEventTicketPCD contains all public/revealed fields.
+ */
+export interface ZKEdDSAEventTicketPCDClaim {
+  partialTicket: Partial<ITicketData>;
+  watermark: string;
+  signer: EDdSAPublicKey; // in montgomery form. must use F.toObject() from ffjavascript to convert to raw coords
+
+  // only if requested in PCDArgs
+  validEventIds?: string[];
+  externalNullifier?: string;
+  nullifierHash?: string;
+}
+
+/**
+ * Proof part of a ZKEdDSAEventTicketPCD is a snarkjs proof.
+ */
+export interface ZKEdDSAEventTicketPCDProof {
+  // TODO: Can this be imported from somewhere?
+  pi_a: [string, string, string];
+  pi_b: [[string, string], [string, string], [string, string]];
+  pi_c: [string, string, string];
+  protocol: string;
+  curve: string;
```

Comment:
@cedoor, re: #465, will we have a type for the snarkjs proof?

Reply:
Sure!